Code/GraphMol/FileParsers/MolFileWriter.cpp

```diff
@@ -38,6 +38,8 @@ namespace RDKit {
 //*************************************
 
 namespace {
+const std::string writeAtomList = "_writeAtomList";
```

Comment:
Should we move writeAtomList into the standard string set in RDGeneral/types.h?

Reply:
Ok, writeAtomList appears to be a temporary value that is set to control how the atom list is written.  And it is temporary because 
 
1.  that nasty logic to see if we are Q or L.  and
2. we only pass a mol object around.

Now, this makes the mol writer quite not thread safe which I think is problematic since it takes a const ROMol.  A perhaps another solution is to make a companion data object:

```
struct MolData{
   dynamic_bitset<> writeAtomList;
}

const std::string GetMolFileQueryInfo(const RWMol &mol, MolData &data) {
   std::stringstream ss;
   boost::dynamic_bitset<> listQs(mol.getNumAtoms());
   if (hasListQuery(atom) && data.writeAtomList[atom.GetIdx()] ) { ... }
}

...

This could also replace the listQs bitset as well?
```